float32 width
float32 speed
float32 force
int32   grasp_state
